### ISSUE TITLE
Some questions about reproducing the results of PC softmax

### ISSUE THREAD

**Issue by likanchuan** [hyperconnect/LADE#17]
Hi, thanks for sharing the code of your great work. I am interested in your work and inspired a lot by it. When reproducing the PC softmax result on Cifar100-LT with imbalance factor 100, a gap appeared between result I reproduced and those reported in your paper.  

I implemented PC softmax by using the PriorCEloss.py for training and it got an acc of 41.80. Then I changed to train by CE loss and directly add -log p_s(y) to the logits when predicting, and the acc got better to 42.3 but still lower than the result of 45.3 in the paper. I want to ask if you used other methods together with PC softmax or if I misunderstood the PC softmax to implement it wrongly.Or is there other reasons to the gap?

I will appreciate it getting some advice or hints from you. Thanks for sharing your awesome work again.

**Comment by juice500ml** [comment]
Thank you for your interest in our paper 😊

To recap, PC Softmax trains exactly the same with the standard vanilla CE + Softmax, and subtract the log prior (log p_s(y)) only in the test phase.

Hence, to obtain the performance of the PC Softmax, you need to train using "[ce.yaml](https://github.com/hyperconnect/LADE/blob/master/config/CIFAR100_LT/ce.yaml#L2-L6)", where the [evaluation code](https://github.com/hyperconnect/LADE/blob/master/run_networks.py#L594-L597) will subtract the log prior for you!

[PriorCELoss](https://github.com/hyperconnect/LADE/blob/master/loss/PriorCELoss.py#L23) adds the log prior when training, so it's used in training [LADE](https://github.com/hyperconnect/LADE/blob/master/config/CIFAR100_LT/lade.yaml#L1-L11).

Long story short, it seems that you were correct on using the CE loss (I suspect you were referring to "ce.yaml"), but you don't need to subtract the log p_s(y) yourself. Then I guess you'll get the numbers in the paper.

**Comment by likanchuan** [comment]
Thanks for your reply and your advice works really. I get reasonable result now.
